src/compiler/core.ts

```diff
@@ -2165,6 +2165,10 @@ namespace ts {
     }
 
     export function fill<T>(length: number, cb: (index: number) => T): T[] {
-        return new Array(length).fill(0).map((_, i) => cb(i));
+        const result = Array<T>(length);

```

Comment:
At least in V8 this causes the array to be considered "holey" and therefore it's not as optimized as packed arrays.
On the other hand repeatedly calling `push` needs to grow the backing store.

I'd say this is fine as is, as this function is only used in a single place in `generateTypes.ts`. So the holey array won't cause too much deoptimization where it's used. The alternative using `push` wouldn't be a problem either because the arrays are unlikely to contain more than a dozen elements.